integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml

```diff
@@ -9,238 +9,215 @@
-          image: {{ .Values.image }}:{{ .Values.imageTag }}
-          imagePullPolicy: {{ .Values.imagePullPolicy }}
-          resources:
-            limits:
-            {{- if .Values.master.resources  }}
-              {{- if .Values.master.resources.limits }}
-                {{- if .Values.master.resources.limits.cpu  }}
-              cpu: {{ .Values.master.resources.limits.cpu }}
-              memory: {{ .Values.master.resources.limits.memory }}
-                {{- end }}
-              {{- end }}
-            requests:
-              {{- if .Values.master.resources.requests }}
-                {{- if .Values.master.resources.requests.cpu  }}
-              cpu: {{ .Values.master.resources.requests.cpu }}
-              memory: {{ .Values.master.resources.requests.memory }}
-                {{- end }}
-              {{- end }}
-            {{- end }}
-          envFrom:
-            - configMapRef:
-                name: {{ template "alluxio.name" . }}-config
-          env:
-            - name: ALLUXIO_WORKER_HOSTNAME
-              valueFrom:
-                fieldRef:
-                  fieldPath: status.hostIP
-          volumeMounts:
-            - name: tz-config
-              mountPath: /etc/localtime
-            - name: alluxio-journal
-              mountPath: /journal
-            - name: {{ template "alluxio.fullname" . }}-format-script
-              mountPath: /opt/alluxio/format-script
-          command: ["/opt/alluxio/format-script/format-master.sh"]
+      - name: journal-chown
+        image: {{ $root.Values.image }}:{{ $root.Values.imageTag }}
+        imagePullPolicy: {{ $root.Values.imagePullPolicy }}
+        securityContext:
+          runAsUser: 0
+        command: ["/bin/chown","-R", {{ $ugi | quote }}, "/journal"]
+        volumeMounts:
+        - name: alluxio-journal
+          mountPath: /journal
+      {{- end}}
       containers:
         - name: alluxio-master
-          image: {{ .Values.image }}:{{ .Values.imageTag }}
-          imagePullPolicy: {{ .Values.imagePullPolicy }}
+          image: {{ $root.Values.image }}:{{ $root.Values.imageTag }}
+          imagePullPolicy: {{ $root.Values.imagePullPolicy }}
+          securityContext:
+            runAsUser: {{ $root.Values.user }}
+            runAsGroup: {{ $root.Values.group }}
+          {{- if $root.Values.master.resources  }}
           resources:
+            {{- if $root.Values.master.resources.limits }}
             limits:
-            {{- if .Values.master.resources  }}
-              {{- if .Values.master.resources.limits }}
-                {{- if .Values.master.resources.limits.cpu  }}
-              cpu: {{ .Values.master.resources.limits.cpu }}
-              memory: {{ .Values.master.resources.limits.memory }}
-                {{- end }}
-              {{- end }}
+              cpu: {{ $root.Values.master.resources.limits.cpu }}
+              memory: {{ $root.Values.master.resources.limits.memory }}
+            {{- end }}
```

Comment:
I changed this part a bit (This applies to all resources too). I think it makes sense the user can very possibly have resource policies in real life environments, so it makes sense to have neither `requests` nor `limits`. But I think it's reasonable that the user should define both cpu and memory if they want to define some quotas. Checking the existence of both `cpu` and `memory` seems a little bit verbose to me.

This is just my personal opinion. Of course there are scenarios where the user only cares about cpu, not memory. It's just I chose the simpler way out. Open to opinions :)

Reply:
sgtm